### ISSUE TITLE
Nuxt.js + PWA - Service Worker errors

### ISSUE THREAD

**Issue by frankspin89** [nuxt/nuxt.js#1108]
Trying to implement PWA module in my nuxt.js install. But getting some error related to service workers.

**Error 1**
ServiceWorker failed to install: ServiceWorker failed to handle event (event.waitUntil Promise rejected)

**Error 2:**
Uncaught (in promise) request-not-cached: an error was thrown by workbox with error code; '`request-not-chached'

URL: https://proposalmate.now.sh/

![error](https://user-images.githubusercontent.com/4480104/28324605-1db43a0e-6bdc-11e7-8e83-0b48901646c5.PNG)

**package.json**

```
{
  "name": "proposalmate",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "@nuxtjs/component-cache": "^1.0.0",
    "@nuxtjs/pwa": "^0.2.0",
    "autoprefixer": "^7.1.2",
    "css-loader": "^0.28.4",
    "cssnano": "^3.10.0",
    "element-theme-default": "^1.3.7",
    "element-ui": "^1.3.7",
    "nuxt": "^1.0.0-alpha.5",
    "postcss-cssnext": "^3.0.2",
    "postcss-hexrgba": "^0.2.1",
    "postcss-import": "^10.0.0",
    "postcss-loader": "^2.0.6",
    "postcss-nested": "^2.0.4",
    "style-loader": "^0.18.2",
    "vue": "2.3.4",
    "vue-server-renderer": "2.3.4",
    "vue-style-loader": "^3.0.1",
    "vue-template-compiler": "2.3.4"
  },
  "scripts": {
    "dev": "nuxt",
    "build": "nuxt build",
    "start": "nuxt start",
    "generate": "nuxt generate",
    "lint": "eslint --ext .js,.vue --ignore-path .gitignore .",
    "precommit": "npm run lint",
    "element": "et -c css/element-variables.css -o css/theme -w"
  },
  "devDependencies": {
    "babel-eslint": "^7.2.3",
    "eslint": "^4.2.0",
    "eslint-config-standard": "^10.2.1",
    "eslint-loader": "^1.9.0",
    "eslint-plugin-html": "^3.1.0",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-node": "^5.1.0",
    "eslint-plugin-promise": "^3.5.0",
    "eslint-plugin-standard": "^3.0.1"
  }
}
```

**nuxt.config.js**
```
module.exports = {
  /*
  ** Headers of the page
  */

  head: {
    title: 'starter',
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      { hid: 'description', name: 'description', content: 'Nuxt.js project' }
    ],
    link: [
      { rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' },
      { rel: 'dns-prefetch', href: 'https://proposalmate.now.sh' }
    ]
  },
  /*
  ** Customize the progress-bar color
  */
  loading: { color: '#3B8070' },

  manifest: {
    name: 'ProposalMate',
    short_name: 'ProposalMate',
    lang: 'en',
    description: 'Create beautiful business proposals',
    theme_color: '#188269',
    background_color: '#000'
  },

  modules: [
    '@nuxtjs/pwa',
    '@nuxtjs/component-cache'
  ],

  render: {
    static: {
      maxAge: '1y',
      setHeaders(res, path) {
        if (path.includes('sw.js')) {
          res.setHeader('Cache-Control', 'public, max-age=0')
        }
      }
    }
  },

  css: [
    { src: '~/css/theme/index.css', lang: 'postcss' },
    { src: '~/css/main.css', lang: 'postcss' }
  ],

  plugins: [
    'plugins/element-ui.js'
  ],
  /*
  ** Build configuration
  */
  build: {

    /*
    ** Run ESLINT on save
    */
    extend(config, ctx) {

      const cssLoader = config.module.rules.find((loader) => loader.test.toString() === '/\\.css$/')
      cssLoader.use.splice(2, 0, {
        loader: 'postcss-loader',
        options: {
          sourceMap: true
        }
      })

      if (ctx.isClient) {
        config.module.rules.push({
          enforce: 'pre',
          test: /\.(js|vue)$/,
          loader: 'eslint-loader',
          exclude: /(node_modules)/
        })
      }
    }

  }
}
```

**Comment by pi0** [comment]
I think the problem is with your deployment as workbox manifest is not updated with latest build hashes, so all responses all 404.

**Comment by philippe-effinet** [comment]
I have the same problem with the nuxt/pwa-template, with no additionnal code in it.

@pi0 Do you have a trick to make the workbox manifest update with latest hashes on npm run generate ?

**Comment by kangsLee** [comment]
Nuxt path does not seem to work in sw.js.
```workboxSW.router.registerRoute('/\/_nuxt\/.*/', workboxSW.strategies.cacheFirst({}), 'GET');```

**Comment by pi0** [comment]
@kangsLee  I don't think problem is with path regex as it's working in many productions :))

@philippe-effinet It should, you can ensure by removing static/manifest.* and running generate again. 

We may need a repo to reproduce the exact error.

**Comment by pi0** [comment]
**UPDATE** It seems broken with latests dependency upgrade 😕  Investigating...

**Comment by pi0** [comment]
Closing in flavor of nuxt-community/modules#75. You can track progress from there.

**Comment by pi0** [comment]
Should be fixed with `workbox@1.1.0`. Demo: https://nuxt-pwa.glitch.me/

**Comment by frankspin89** [comment]
@pi0 Thanks for the awesome work!

I can confirm that everything is working perfect now, including offline!
